drivers/firmware/ti_sci/ti_sci.c

```diff
@@ -0,0 +1,1547 @@
+				       MSG_DEVICE_SW_STATE_ON);
+}
+
+int ti_sci_cmd_idle_device(const struct device *dev, uint32_t dev_id)
+{
+	return ti_sci_set_device_state(dev, dev_id, 0, MSG_DEVICE_SW_STATE_RETENTION);
+}
+
+int ti_sci_cmd_idle_device_exclusive(const struct device *dev, uint32_t dev_id)
+{
+	return ti_sci_set_device_state(dev, dev_id, MSG_FLAG_DEVICE_EXCLUSIVE,
+				       MSG_DEVICE_SW_STATE_RETENTION);
+}
+
+int ti_sci_cmd_put_device(const struct device *dev, uint32_t dev_id)
+{
+	return ti_sci_set_device_state(dev, dev_id, 0, MSG_DEVICE_SW_STATE_AUTO_OFF);
+}
+
+int ti_sci_cmd_dev_is_valid(const struct device *dev, uint32_t dev_id)
+{
+	uint8_t unused;
+
+	return ti_sci_get_device_state(dev, dev_id, NULL, NULL, NULL, &unused);
+}
+
+int ti_sci_cmd_dev_get_clcnt(const struct device *dev, uint32_t dev_id, uint32_t *count)
+{
+	return ti_sci_get_device_state(dev, dev_id, count, NULL, NULL, NULL);
+}
+
+int ti_sci_cmd_dev_is_idle(const struct device *dev, uint32_t dev_id, bool *r_state)
+{
+	int ret;
+	uint8_t state;
+
+	if (!r_state) {
+		return -EINVAL;
+	}
+
+	ret = ti_sci_get_device_state(dev, dev_id, NULL, NULL, &state, NULL);
+	if (ret) {
+		return ret;
+	}
+
+	*r_state = (state == MSG_DEVICE_SW_STATE_RETENTION);
+
+	return 0;
+}
+
+int ti_sci_cmd_dev_is_stop(const struct device *dev, uint32_t dev_id, bool *r_state,
+			   bool *curr_state)
+{
+	int ret;
+	uint8_t p_state, c_state;
+
+	if (!r_state && !curr_state) {
+		return -EINVAL;
+	}
+
+	ret = ti_sci_get_device_state(dev, dev_id, NULL, NULL, &p_state, &c_state);
+	if (ret) {
+		return ret;
+	}
+
+	if (r_state) {
+		*r_state = (p_state == MSG_DEVICE_SW_STATE_AUTO_OFF);
+	}
+	if (curr_state) {
+		*curr_state = (c_state == MSG_DEVICE_HW_STATE_OFF);
+	}
+
+	return 0;
+}
+
+int ti_sci_cmd_dev_is_on(const struct device *dev, uint32_t dev_id, bool *r_state, bool *curr_state)
+{
+	int ret;
+	uint8_t p_state, c_state;
+
+	if (!r_state && !curr_state) {
+		return -EINVAL;
+	}
+
+	ret = ti_sci_get_device_state(dev, dev_id, NULL, NULL, &p_state, &c_state);
+	if (ret) {
+		return ret;
+	}
+
+	if (r_state) {
+		*r_state = (p_state == MSG_DEVICE_SW_STATE_ON);
+	}
+	if (curr_state) {
+		*curr_state = (c_state == MSG_DEVICE_HW_STATE_ON);
+	}
+
+	return 0;
+}
+
+int ti_sci_cmd_dev_is_trans(const struct device *dev, uint32_t dev_id, bool *curr_state)
+{
+	int ret;
+	uint8_t state;
+
+	if (!curr_state) {
+		return -EINVAL;
+	}
+
+	ret = ti_sci_get_device_state(dev, dev_id, NULL, NULL, NULL, &state);
+	if (ret) {
+		return ret;
+	}
+
+	*curr_state = (state == MSG_DEVICE_HW_STATE_TRANS);
+
+	return 0;
+}
+
+int ti_sci_cmd_set_device_resets(const struct device *dev, uint32_t dev_id, uint32_t reset_state)
+{
+	struct ti_sci_msg_req_set_device_resets req;
+	struct ti_sci_msg_hdr *resp;
+	struct ti_sci_xfer *xfer;
+	int ret = 0;
+
+	xfer = ti_sci_setup_one_xfer(dev, TI_SCI_MSG_SET_DEVICE_RESETS,
+				     TI_SCI_FLAG_REQ_ACK_ON_PROCESSED, (uint32_t *)&req,
+				     sizeof(req), sizeof(*resp));
+	if (!xfer) {
```

Comment:
why are these checks again inconsistent? on all other instances you don't have this.